deeppavlov/core/data/dataset.py

```diff
@@ -34,24 +51,35 @@ def __init__(self, data: Dict[str, List[Tuple[Any, Any]]], seed: int = None, *ar
             'all': self.train + self.test + self.valid
         }
 
-    def batch_generator(self, batch_size: int, data_type: str = 'train') -> Generator:
-        r"""This function returns a generator, which serves for generation of raw (no preprocessing such as tokenization)
+    def batch_generator(self, batch_size: int, data_type: str = 'train',
+                        shuffle: bool = None, seed: int = None) -> Generator:
+        r"""This function returns a generator, which serves for generation of raw
+        (no preprocessing such as tokenization)
          batches
         Args:
             batch_size (int): number of samples in batch
             data_type (str): can be either 'train', 'test', or 'valid'
+            shuffle (bool): whether to shuffle dataset before batching
+            seed (int): random seed for batching
         Returns:
             batch_gen (Generator): a generator, that iterates through the part (defined by data_type) of the dataset
         """
+        if shuffle is not None:
+            self.shuffle = shuffle
```

Comment:
I don't think that batch generator should be able to switch the shuffle flag for the dataset forever